### PR TITLE
Add shared configs

### PR DESCRIPTION
**Pull request by narwold**
## PRs

**Add shared configs option to labeler action (this PR)**
[Add shared labels to .github](https://github.com/beamtech/.github/pull/16)
[Add labels to `authentication-server`](https://github.com/beamtech/authentication-server/pull/465)
[Add labels to `js-frontend`](https://github.com/beamtech/js-frontend/pull/3941)

## Context

This PR adds the ability for the `labeler` action to accept remote shared configs so that org-wide label patterns can be applied to any given repo.

## Validation

Validate with https://github.com/beamtech/authentication-server/pull/465

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -14534,49 +14525,60 @@ const core = __importStar(__webpack_require__(186));
...
+        const { data: pullRequest } = await client.pulls.get({
+            owner: github.context.repo.owner,
+            repo: github.context.repo.repo,
+            pull_number: prNumber,
+        });
+        core.debug(`fetching changed files for pr #${prNumber}`);
+        const changedFiles = await getChangedFiles(client, prNumber);
+        const localLabelGlobs = await getLabelGlobs(client, configPath);
+        const sharedConfigGlobs = await Promise.all(sharedConfigurations.map((config) => {
+            const [, repoOwner, repoName, repoPath] = config.match(/([^/]+)\/([^/]+)\/?(.*)?@/);
```
Comment (jnarwold) [inline dist/index.js]: What's the logic behind the initial comma in the array

Comment (narwold) [reply dist/index.js]: A `match` result always includes the FULL match as the first arg, followed by each of the capturing groups in the regex. I only want the capturing groups, so I want to skip and not use the first arg. Including it in the destructuring means the linter will complain that it's unused, so I learned a while back that this syntax is allowed and you can imply that it's there with the extra comma but not assign it to a variable name.

Comment (silasb) [reply dist/index.js]: Typically languages use `_` like so `[_, variable_i_want]` to give more control. @narwold with JS can you do the empty variable anywhere within the expanded array list?  Say the 2nd or 3rd argument?

Comment (narwold) [reply dist/index.js]: `_` in JS will actually create a variable, and the linter will complain about that too. Yes, you can do it in any arg.

File: tsconfig.json [hunk 2]
```diff
@@ -2,17 +2,17 @@
   "compilerOptions": {
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
-    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
-    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "target": "ES2019" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */,
```
Comment (silasb) [inline tsconfig.json]: Why bump this up to ES2019?

Comment (narwold) [reply tsconfig.json]: I'll switch it back and test to remember. There was some syntax I was using that barked and said I needed 2019.

File: src/main.ts [hunk 3]
```diff
@@ -8,12 +8,23 @@ interface MatchConfig {
   any?: string[];
 }
 
+type RemoteRepoDetails = {
```
Comment (silasb) [inline src/main.ts:11]: Is there anything specific here for Beam?  I can't seem to see any code that would be specific.  Why not open a PR up against against the forked project?

Comment (narwold) [reply src/main.ts:11]: I can, and I probably should, but I figured I'd get into a big discussion about recommended API and all that and it probably wouldn't be a super-simple change from their perspective... but yeah, you're right.

File: tsconfig.json (hunk 2 above)
Comment (narwold) [reply tsconfig.json]: It's `flat` I was trying to use

File: src/main.ts (hunk 3 above)
Comment (narwold) [reply src/main.ts:11]: I'm not sure how long the OSS process will take, so I'd rather that be a secondary PR so we can start using this. I think that was the thought.

Comment (silasb) [reply src/main.ts:11]: I say try it.  I think all these changes make sense for this labeler. Can `shared-configuration` be empty and still work, is this backwards compatible?

File: tsconfig.json (hunk 2 above)
Comment (silasb) [reply tsconfig.json]: 👍

Comment (silasb) [reply tsconfig.json]: This bump in the target is also the reason why the `dist/index.js` file changed so much?

Comment (narwold) [reply tsconfig.json]: Probably. When I get back later, I'll try to use `reduce` instead.

File: src/main.ts (hunk 3 above)
Comment (narwold) [reply src/main.ts:11]: Yes, I believe so. I'll double-check/test for that again.

**Review by silasb — APPROVED** [review]
Looks good.  Going to issue a PR against the upstream repo to see if we can avoid a forked version.

**Comment by narwold** [comment]
> Looks good. Going to issue a PR against the upstream repo to see if we can avoid a forked version.

https://github.com/actions/labeler/pull/141

**Pull request merged**